### PR TITLE
update CONTRIBUTING and README documentation

### LINKED ISSUE #1299: Add a note to CONTRIBUTING.md about m1 macs
### Feature Description

Currently the EE container for GitLab has issues running on m1 macs because there is no native ARM container. We should add a note to our contributing file noting we're aware of this, and we suggest using GitPod to work around it.

### Do you want to implement this?

- [X] I would like to implement this myself 👷

### PR DESCRIPTION
**Pull request by PatrickRice-KSC**
## Description

This PR adds a minor note to the CONTRIBUTING.md about the known issue with the m1 development evnrionment (and assumes the same is true of the m2 processor since it's likely an ARM issue).

I also took this opportunity to update the discord link in our README.md.

Closes #1299

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -107,6 +107,11 @@ terraform {
 }
 ```
 
+### M1/M2 Environments
+
+The apple silicon environments currently have an issue where the local GitLab container will now start to run tests. We recommend that you use GitPod for developing the
```
Comment (timofurrer) [inline CONTRIBUTING.md:112]:
```suggestion
The apple silicon environments currently have an issue where the local GitLab container will not start to run tests. We recommend that you use GitPod for developing the 
```

**Review by timofurrer — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (PatrickRice-KSC) [reply CONTRIBUTING.md]: nice catch, thanks!

**Review by Shocktrooper — DISMISSED** [review]

**Comment by timofurrer** [comment]
Could you please rebase onto the latest main (see #1304)?

**Review by Shocktrooper — APPROVED** [review]

**Comment by Shocktrooper** [comment]
@timofurrer I think you need to mark your request as done assuming there are no more changes to make before this can be merged

**Review by timofurrer — APPROVED** [review]

**Pull request merged**